proxy/http2/HTTP2.cc

```diff
@@ -434,17 +437,18 @@ http2_convert_header_from_2_to_1_1(HTTPHdr *headers)
     }
 
     // Parse URL
-    Arena arena;
-    size_t url_length     = scheme_len + 3 + authority_len + path_len;
-    char *url             = arena.str_alloc(url_length);
-    const char *url_start = url;
-
-    memcpy(url, scheme, scheme_len);
-    memcpy(url + scheme_len, "://", 3);
-    memcpy(url + scheme_len + 3, authority, authority_len);
-    memcpy(url + scheme_len + 3 + authority_len, path, path_len);
-    url_parse(headers->m_heap, headers->m_http->u.req.m_url_impl, &url_start, url + url_length, true);
-    arena.str_free(url);
+    {
+      size_t url_length = scheme_len + 3 + authority_len + path_len;
+      ts::LocalBuffer local_buffer(url_length);
+      char *url             = reinterpret_cast<char *>(local_buffer.get());
+      const char *url_start = url;
+
+      memcpy(url, scheme, scheme_len);
+      memcpy(url + scheme_len, "://", 3);
+      memcpy(url + scheme_len + 3, authority, authority_len);
+      memcpy(url + scheme_len + 3 + authority_len, path, path_len);
+      url_parse(headers->m_heap, headers->m_http->u.req.m_url_impl, &url_start, url + url_length, true);
```

Comment:
It looks like we can avoid this memcpy & parse by setting scheme, host, and path directly.

Reply:
Done by https://github.com/apache/trafficserver/pull/6545